docs/user-installation-review.md

```diff
@@ -0,0 +1,211 @@
+
+The steps below use a utility `insights-ocp-etcd-logs` to parse the logs, aggregate the requests into buckets of 100ms from 200ms to 1s and report it on the stdout.
+
+This is the utility to help you to troubleshoot the slow requests in your cluster, and help to take some decisions like changing the flavor of the block device used by the control plane, increasing IOPS, changing the flavor of the instances, etc.
+
+There's no magic or desired number, but for reference, based on the observations from integrated platforms, is to have no more than 30-40% of requests above 500ms while running the certification tests.
+
+> TODO: provide guidance on how to get the errors from the etcd pods, and parse it into buckets of latency to understand the performance of the etcd while running the certification environment.
+
+- Export the location you must-gather has been extracted:
+
+```bash
+export MUST_GATHER_PATH=${PWD}/must-gather.local.2905984348081335046
+```
+
+- Extract the utility from the tools repository:
+
+> This binary will be available when this card will be completed: https://issues.redhat.com/browse/SPLAT-857
+
+```bash
+TOOLS_IMAGE=$(skopeo inspect docker://quay.io/ocp-cert/tools:latest | jq .Digest)
+oc image extract ${TOOLS_IMAGE} --file="/usr/bin/insights-ocp-etcd-logs"
+chmod u+x insights-ocp-etcd-logs
+```
+
+- Overall report:
+
+> Note: This report can not be usefull depending how old is the logs. We recommend looking at the next report which aggregates by the hour, so you can check the time frame the certification has been executed
+
+> More information about the utility: https://issues.redhat.com/browse/SPLAT-857
+
+```bash
+grep -rni "apply request took too long" ${MUST_GATHER_PATH} \
+    | grep -Po 'took":"([a-z0-9\.]+)"' \
+    | awk -F'took":' '{print$2}' \
+    | tr -d '"' \
+    | ./insights-ocp-etcd-logs
+```
+
+- Report aggregated by hour:
+
+```bash
+FILTER_MSG="apply request took too long"
+for TS in $( grep -rni "${FILTER_MSG}" ${MUST_GATHER_PATH} \
+    | awk '{print$1}' \
+    | awk -F'.log:' '{print$2}' \
+    | awk -F':' '{print$2}' \
+    | sort | uniq); do
+    echo "-> ${TS}"
+    grep -rni "${FILTER_MSG}" ${MUST_GATHER_PATH} \
+        | grep $TS \
+        | grep -Po 'took":"([a-z0-9\.]+)"' \
+        | awk -F'took":' '{print$2}' \
+        | tr -d '"' \
+        | ./insights-ocp-etcd-logs
+done
+```
+
+#### Mount /var/lib/etcd in sepparate disk <a name="components-etcd-mount"></a>
+
+One way to improve the performance on etcd is to use a dedicated block device.
+
+You can mount `/var/lib/etcd` by following the documentation:
+
+- [OpenShift Docs: Disk partitioning](https://docs.openshift.com/container-platform/4.11/installing/installing_bare_metal/installing-bare-metal.html#installation-user-infra-machines-advanced_disk_installing-bare-metal)
+- [KCS: Mounting separate disk for OpenShift 4 etcd](https://access.redhat.com/solutions/5840061)
+
+### Image Registry <a name="components-imageregistry"></a>
```

Comment:
I'm afraid this information is redundant with the e2e tests. If the image registry does not function properly or you cannot push/build/pull images then the e2e tests will fail. There's no grey area with the image registry like there is with etcd performance so I'm not entirely sure we still need this section. In other words, why have the user manually repeat tests that will be run in the e2e tests.

Reply:
The point to add this here is to make sure the partner tested it before taking time running, collecting data, investigating, sending it to support.. As this is a *helper* document, it was created as a *reminder*.

wdyt?

Reply:
I append this:

> (...) otherwise, the e2e tests will be reported as failed.

to the line

> You should be able to access the registry and make sure you can push and pull images on it, otherwise, the e2e tests will be reported as failed.